src/fastoad/openmdao/problem.py

```diff
@@ -198,21 +198,44 @@ def _get_problem_inputs(self) -> Tuple[VariableList, VariableList]:
 
         :return: VariableList of needed input variables, VariableList with unused variables.
         """
+
         problem_inputs_names = [var.name for var in self.analysis.problem_variables if var.is_input]
 
-        input_variables = DataFile(self.input_file_path)
+        input_file_variables = DataFile(self.input_file_path)
 
         unused_variables = VariableList(
-            [var for var in input_variables if var.name not in problem_inputs_names]
+            [var for var in input_file_variables if var.name not in problem_inputs_names]
         )
         for name in unused_variables.names():
-            del input_variables[name]
+            del input_file_variables[name]
+
+        non_filled_variable_names = self._get_remaining_nan_variable_names(input_file_variables)
+
+        if non_filled_variable_names:
+            raise FASTOpenMDAONanInInputsError(self.input_file_path, non_filled_variable_names)
 
-        nan_variable_names = [var.name for var in input_variables if np.any(np.isnan(var.value))]
-        if nan_variable_names:
-            raise FASTOpenMDAONanInInputFile(self.input_file_path, nan_variable_names)
+        return input_file_variables, unused_variables
 
-        return input_variables, unused_variables
+    def _get_remaining_nan_variable_names(self, input_file_variables):
```

Comment:
Based on the docstring and the way `non_filled_variable_names` is built, it seems like this function return a list of problem inputs whose default is nan and who do not appear in the input of that are set at nan in the input file. in that case, why does the function name contains `reamining` ? It seems like it get all NaN variables

Reply:
The idea is getting the variables that "remain" at NaN after reading the input file (which match your description, I guess).
But I would gladfully consider another naming...

Reply:
I do understand the choice of the word remaining now. I don't have any name that are more explicit in mind, maybe simply `_get_nan_variable_names` ?